docs/src/tasks.md

```diff
@@ -132,6 +132,19 @@ Or explicitly include escaped quotes like so:
 }
 ```
 
+### Task filtering based on variables
+
+Zed filters tasks based on what is currently able to be run. To do this, Zed will consider the presence or absence of environment variables. For example, the following task is only available to be run when text is selected in the editor.
+
+```json
+{
+  "label": "selected text",
+  "command": "echo \"$ZED_SELECTED_TEXT\""
+}
+```
+
+If you would like to define a task that works both when text is selected and when text is not selected, you need to create 2 tasks, one with `$ZED_SELECTED_TEXT` present and one without.
```

Comment:
```suggestion
Set default values to such variables to have such tasks always displayed:

{
  "label": "selected text",
  "command": "echo \"${ZED_SELECTED_TEXT:no text selected}\""
}
```

(GH markdown did not allow nested code blocks, so please add those)


This is quite an odd sentence, or I do not understand the context of https://github.com/zed-industries/zed/issues/38614#issuecomment-3318038681 then.
Let's discuss more what do you want to achieve, if this is not the solution; otherwise let's rewrite the paragraph.

Reply:
Using shell expansion defaults like you suggested do not work. My understanding of what Zed is doing is simply parsing the task as a string and searching for environmental variable names as a substring.

Before I opened #38614, I had tried every possible combination of shell expansion of the variables, including subshells, echo, printf. I had also tried including `$ZED_SELECTED_TEXT` in the `env`, `command`, and `args` fields in the task spec, and no combination would show the task when text is not selected.

At the moment, one does need to create 2 separate tasks, one with and one without `$ZED_SELECTED_TEXT` in order to have the task work when text is selected or not.